.changeset/selfish-hounds-do.md

```diff
@@ -0,0 +1,5 @@
+---
+'@compiled/react': patch
+---
+
+Class name component now has its style prop passed through children as function typed as `CSSProperties` instead of a string map.
```

Comment:
I would just be more concise, and use the component name directly:

`` `ClassNames` component now has its style prop typed as `CSSProperties` instead of a string map``